lib/decanter/parser/boolean_parser.rb

```diff
@@ -1,12 +1,15 @@
+# frozen_string_literal: true
+
 module Decanter
   module Parser
     class BooleanParser < ValueParser
-
       allow TrueClass, FalseClass
 
-      parser do |val, options|
-        next if (val.nil? || val === '')
-        [1, '1'].include?(val) || !!/^true$/i.match?(val.to_s)
+      parser do |val, _options|
+        next if val.nil? || val == ''
+
+        true_values = [1, '1', true, 'true', 'TRUE', 'True']
+        true_values.include?(val) || val.to_s.downcase == 'true'
```

Comment:
Non-blocking but there might be some additional refactoring opportunities in this block. If we downcase the val (if it's a string), we don't have to check both `'TRUE'` and `'true'`.